src/assets/Azure.Core.Shared/PageResponseEnumerator.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    internal static class PageResponseEnumerator
```

Comment:
I don't see this file added to the copy sources script.

Reply:
You cannot download this file as I've changed it from the original source. The original class isn't partial, or else I would have downloaded the original and then added other methods to it.

Reply:
Do we only need partial for the static helper method? I would rather not modify shared source files too much. Or do it in an automated fashion.

Reply:
I would think that partial on any shared source would make sense, right? If it is shared source, anyone using it should be able to add to it. In this case, I initially wanted to add overloads for the methods already there. Now, I've very much changed the source (classes and originally methods are different), so I couldn't use partial anyway.

Reply:
Add a find-replace to mirroring script like this https://github.com/Azure/autorest.csharp/blob/feature/v3/eng/DownloadSharedSource.ps1#L13 so it's automatic

Reply:
So, compare these 2 files side-by-side:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Shared/PageResponseEnumerator.cs

https://github.com/MiYanni/autorest.csharp/blob/paging-support/src/assets/Azure.Core.Shared/PageResponseEnumerator.cs

You'll see that the public methods have the same interface. But, everything internal to how they work has changed. The classes operate with 2 funcs instead of 1. And, I added additional public methods that was used by the generator. Do you think it makes sense to name this new version a different name? Name the classes differently? Basically, this new file doesn't have anything to do with the original, other than the original 4 public methods having the same interface. But, I felt it is 'in the spirit of' the original, just enhanced to allow 2 funcs to define a paging instead of 1.

Reply:
I didn't even notice that. Let me review the changes.

Reply:
Rename the file it's very confusing otherwise.

Reply:
For now, I've renamed the class itself (and fixed a bug). I'll rename the file in the branch I have with all the other file renaming. Otherwise, it'll make merging more confusing.

Reply:
Actually, I'm not renaming this now as it will cause issues with file name conflicts. I'm simply going to put the bug fix in. I'll rename in the next PR.